cirq/protocols/mixture_protocol.py

```diff
@@ -82,63 +83,6 @@ def mixture(val: Any, default: Any = RaiseTypeErrorIfNotProvided
         with that probability in the mixture. The probabilities will sum to 1.0.
     """
 
-    getter = getattr(val, '_mixture_', None)
-    result = NotImplemented if getter is None else getter()
-
-    if result is not NotImplemented:
-        return result
-    if default is not RaiseTypeErrorIfNotProvided:
-        return default
-
-    if getter is None:
-        raise TypeError("object of type '{}' has no _mixture_ method.".format(
-            type(val)))
-
-    raise TypeError("object of type '{}' does have a _mixture_ method, "
-                    "but it returned NotImplemented.".format(type(val)))
-
-
-def has_mixture(val: Any) -> bool:
-    """Returns whether the value has a mixture representation.
-
-    Returns:
-        If `val` has a `_has_mixture_` method and its result is not
-        NotImplemented, that result is returned. Otherwise, if the value
-        has a `_mixture_` method return True if that has a non-default value.
-        Returns False if neither function exists.
-    """
-    getter = getattr(val, '_has_mixture_', None)
-    result = NotImplemented if getter is None else getter()
-    if result is not NotImplemented:
-        return result
-
-    # No _has_mixture_ function, use _mixture_ instead
-    return mixture(val, None) is not None
-
-
-def mixture_channel(val: Any, default: Any = RaiseTypeErrorIfNotProvided
-                   ) -> Sequence[Tuple[float, np.ndarray]]:
-    """Return a sequence of tuples for a channel that is a mixture of unitaries.
-
-    In contrast to `mixture` this method falls back to `unitary` if `_mixture_`
-    is not implemented.
-
-    A mixture channel is described by an iterable of tuples of the form
-
-        (probability of unitary, unitary)
-
-    The probability components of the tuples must sum to 1.0 and be between
-    0 and 1 (inclusive) and the `unitary` must be a unitary matrix.
-
-    Args:
-        val: The value whose mixture_channel is being computed.
-        default: A default value if val does not support mixture.
-
-    Returns:
-        An iterable of tuples of size 2. The first element of the tuple is a
-        probability (between 0 and 1) and the second is the unitary that occurs
-        with that probability. The probabilities will sum to 1.0.
-    """
     mixture_getter = getattr(val, '_mixture_', None)
```

Comment:
not sure why it let me, but the docstring needs to be update to move away from the generic (prob, object) and to the specific (prob, unitary).

Reply:
Done.